streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -344,40 +343,41 @@ public void resume() {
     }
 
     /**
+     * @throws StreamsException fatal error that should cause the thread to die
+     * @throws TaskMigratedException recoverable error that would cause the task to be removed
```

Comment:
I think it is a good habit keeping that since those functions are called by various callers and if the caller do not catch them, the the ancestor callers should -- I've personally found such debugging very helpful.

Reply:
Works for me -- I am just "worried" that those comments might get out-of-sync with the code at some point leading to bug, as we rely on the comments when changing the caller code...